app/Models/Plugin.php

```diff
@@ -0,0 +1,393 @@
+<?php
+
+namespace App\Models;
+
+use App\Contracts\Plugins\HasPluginSettings;
+use App\Enums\PluginCategory;
+use App\Enums\PluginStatus;
+use App\Facades\Plugins;
+use Exception;
+use Filament\Schemas\Components\Component;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Str;
+use JsonException;
+use Sushi\Sushi;
+
+/**
+ * @property string $id
+ * @property string $name
+ * @property string $author
+ * @property string $version
+ * @property string|null $description
+ * @property PluginCategory $category
+ * @property string|null $url
+ * @property string|null $update_url
+ * @property string $namespace
+ * @property string $class
+ * @property string|null $panels
+ * @property string|null $panel_version
+ * @property string|null $composer_packages
+ * @property PluginStatus $status
+ * @property string|null $status_message
+ * @property int $load_order
+ */
+class Plugin extends Model implements HasPluginSettings
+{
+    use Sushi;
```

Comment:
While we can keep using Sushi, I believe Filament v4 already allows non Laravel models. So if Sushi is only used for that, then we should consider removing the Sushi requirement IMO.